rust/gui-client/src-tauri/src/client/gui/os_linux.rs

```diff
@@ -1,6 +1,34 @@
-use anyhow::Result;
+use anyhow::{Context as _, Result};
 use tauri::api::notification::Notification;
 
+pub(crate) async fn set_autostart(enabled: bool) -> Result<()> {
```

Comment:
Not a huge fan of functions with boolean parameters. What if we just make two instead?

Reply:
That would fix the problem with the weird if-else chain, but then I'd want a third function returning `Result<(PathBuf, PathBuf)>` to compute `dir` and `link`, and somewhere up top the CLI / GUI will have to translate a bool into `enable` and `disable`.

Could we do an enum, `AutoStart::Enabled` and `AutoStart::Disabled`?

Reply:
> and somewhere up top the CLI / GUI will have to translate a bool into `enable` and `disable`.

Or you have two commands? I am not really fussed either way but I do think there should be two functions because the current one basically doesn't share any code other than computing the path.

I am assuming the commands are not going to be around for long and we will eventually make a check-box in the settings for autostart?

Reply:
Yeah eventually it'll be replaced with a checkbox and I figured the JS will pass a boolean or enum to Rust

Reply:
> Yeah eventually it'll be replaced with a checkbox and I figured the JS will pass a boolean or enum to Rust

Makes sense! :)